### ISSUE TITLE
A-Client - upload to anaconda-package with private flag support

### ISSUE THREAD

**Issue by stephenakearns** [Anaconda-Server/anaconda-client#288]
If you want to upload a private package... 

Current: (annoyingly 2 steps) 
- create empty anaconda-package with `anaconda create`
- `anaconda upload` with the --private switch

The client supports creating a new package in one step with `anaconda upload` by creating the package if it doesn't already exist, but it's not smart enough to do this if the private flag is on. This needs to be changed so that it still respects the --private flag and creates a private package (assuming you have permissions and the right subcription level to do it)

If you don't have the right subscription level, it should provide a helpful warning message that suggests upgrading your account in order to upload private packages and provides a url to `https://anaconda.org/<USER_NAME>/settings/billing/plan`

source: @hsparra

**Comment by srossross** [comment]
@stephenakearns there was a bigger discussion about this last year. We have already moved away from a single step upload of private packages. We did this for a few reasons. 1) You only need to perform the refister step once, not before every upload 2) This is a standard way of doing things, pypi, npm etc require this step 3) Users were confused about uploading private packages since the registration only occurs on the first upload. the `--private` flag would be ignored on subsequent uploads.
